jbpm-document/src/main/java/org/jbpm/document/Documents.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.document;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * A collection of {@link Document Documents}.
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "documents-object")
+public class Documents implements Serializable {
```

Comment:
so in the end you didn't go for list of documents? Was it because of the type erasure? Or you don't see it as valuable option? Thinking on how this would work with forms? Maybe this approach will be actually better...

Reply:
Yes, type erasure in Java does not allow to properly determine the generic type of a List (or Collection in general). So, we would need to check at runtime the class type of elements of the collection (and even that does not give you a guarantee). Second, with an empty list you would probably need to dive into reflection.

This wrapper seems to be a cleaner solution.